app/models/link.rb

```diff
@@ -5,8 +5,10 @@
 ###
 
 class Link < ApplicationRecord
-  scope :for, ->(location) do
-    where(location: location)
+  def self.for(location)
+    @cache ||= {}
+
+    @cache[location] ||= where(location: location)
```

Comment:
These do change (there's an interface for adding menu items)  Not 100% sure it's the same items, I'm not remembering the `:for` scope, butt we should make sure to invalidate this cache if the items are added/removed.